### ISSUE TITLE
invalid or unknown key: filter

### ISSUE THREAD

**Issue by saikrishnaburugula** [terraform-providers/terraform-provider-aws#5133]
Hello,

I'm using terraform v0.11.3 version, but assigning an elastic ip address to an Ec2 instance using data filter or tags is not supported with this version. It is giving me invalid or unknown key when using filter or tags in the query to look at the elastic ip's.

### Terraform Version
Terraform v0.11.3
+ provider.aws v1.25.0
+ provider.template v1.0.0

### Affected Resource(s)
aws_eip_association
data.aws_eip

### Terraform Configuration Files
data "aws_eip" "example" {
  filter {
    name   = "tag:Name"
    values = ["test"]
  }
}
resource "aws_eip_association" "eip_assoc" {
  instance_id = "${aws_instance.test.id}"
  allocation_id = "${data.aws_eip.test.id}"
}

### Expected Behavior

terraform apply

### Actual Behavior
Error: data.aws_eip.test: : invalid or unknown key: filter

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Looks like the issue is fixed in some version but not supporting 
$ terraform --version
Terraform v0.11.3
+ provider.aws v1.25.0
+ provider.template v1.0.0

### References

* #0000

**Comment by kl4w** [comment]
PR's submitted (with some overlaps):
* https://github.com/terraform-providers/terraform-provider-aws/pull/3525
* https://github.com/terraform-providers/terraform-provider-aws/pull/5131
* https://github.com/terraform-providers/terraform-provider-aws/pull/3505

**Comment by porschberg** [comment]
The same for Terraform v0.11.8 .

**Comment by praveenprem** [comment]
I'm seeing the same error for **aws_route_tables** as well when I used with Data Source.

### Terraform Version
Terraform v0.11.8
+ provider.aws v1.25.0
+ provider.external v1.0.0
+ provider.null v1.0.0
+ provider.template v1.0.0

### Terrafrom Config
```
data "aws_route_tables" "aurora_cluster_rtbs" {
  vpc_id = "${var.data_vpc}"

  filter {
    name = "tag:Name"
    values = [ "data-aurora.aurora_subnet" ]
  }
}
```

### Behavior
```
➜  infrastructure git:(develop) ✗ terraform plan

Error: data.aws_route_tables.aurora_cluster_rtbs: : invalid or unknown key: filter
```

Is there any workaround for this?

**Comment by bflad** [comment]
Hi folks 👋 Very sorry for the lengthy delays with merging this support, there were a few conflicting pull requests to the same code which needed to get sorted out. Good news is that the following will be supported in version 1.44.0 of the AWS provider, likely releasing later today or tomorrow:
* Search by `filter`
* Search by `tags`
* EC2-Classic support

@praveenprem it looks like `aws_route_tables` data source support for `filter` was added via #5035 which was released in version 1.26.0 of the AWS provider.
